drivers/scheduler/k8s/k8s.go

```diff
@@ -545,80 +617,14 @@ func (k *k8s) WaitForRunning(ctx *scheduler.Context) error {
 }
 
 func (k *k8s) Destroy(ctx *scheduler.Context, opts map[string]bool) error {
-	k8sOps := k8s_ops.Instance()
 	var podList []v1.Pod
 	for _, spec := range ctx.App.SpecList {
-		if obj, ok := spec.(*apps_api.Deployment); ok {
-			if value, ok := opts[scheduler.OptionsWaitForResourceLeakCleanup]; ok && value {
-				if pods, err := k8sOps.GetDeploymentPods(obj); err != nil {
-					logrus.Warnf("[%v] Error getting deployment pods. Err: %v", ctx.App.Key, err)
-				} else {
-					podList = append(podList, pods...)
-				}
-			}
-			t := func() (interface{}, bool, error) {
-				err := k8sOps.DeleteDeployment(obj.Name, obj.Namespace)
-				if err != nil {
-					return nil, true, err
-				}
-				return nil, false, nil
-			}
-
-			_, err := task.DoRetryWithTimeout(t, k8sDestroyTimeout, defaultRetryInterval)
-			if err != nil {
-				return &scheduler.ErrFailedToDestroyApp{
-					App:   ctx.App,
-					Cause: fmt.Sprintf("Failed to destroy Deployment: %v. Err: %v", obj.Name, err),
-				}
-			}
-
-			logrus.Infof("[%v] Destroyed deployment: %v", ctx.App.Key, obj.Name)
-		} else if obj, ok := spec.(*apps_api.StatefulSet); ok {
-			if value, ok := opts[scheduler.OptionsWaitForResourceLeakCleanup]; ok && value {
-				if pods, err := k8sOps.GetStatefulSetPods(obj); err != nil {
-					logrus.Warnf("[%v] Error getting statefulset pods. Err: %v", ctx.App.Key, err)
-				} else {
-					podList = append(podList, pods...)
-				}
-			}
-			t := func() (interface{}, bool, error) {
-				err := k8sOps.DeleteStatefulSet(obj.Name, obj.Namespace)
-				if err != nil {
-					return nil, true, err
-				}
-				return nil, false, nil
-			}
-
-			_, err := task.DoRetryWithTimeout(t, k8sDestroyTimeout, defaultRetryInterval)
-			if err != nil {
-				return &scheduler.ErrFailedToDestroyApp{
-					App:   ctx.App,
-					Cause: fmt.Sprintf("Failed to destroy stateful set: %v. Err: %v", obj.Name, err),
-				}
-			}
-
-			logrus.Infof("[%v] Destroyed StatefulSet: %v", ctx.App.Key, obj.Name)
-		} else if obj, ok := spec.(*v1.Service); ok {
-			t := func() (interface{}, bool, error) {
-				err := k8sOps.DeleteService(obj.Name, obj.Namespace)
-				if err != nil {
-					return nil, true, err
-				}
-				return nil, false, nil
-			}
-
-			_, err := task.DoRetryWithTimeout(t, k8sDestroyTimeout, defaultRetryInterval)
-			if err != nil {
-				return &scheduler.ErrFailedToDestroyApp{
-					App:   ctx.App,
-					Cause: fmt.Sprintf("Failed to destroy Service: %v. Err: %v", obj.Name, err),
-				}
-			}
-
-			logrus.Infof("[%v] Destroyed Service: %v", ctx.App.Key, obj.Name)
+		pods, err := k.destroyCoreObject(spec, opts, ctx.App)
```

Comment:
Invoke this wrapped by `task.DoRetryWithTimeout` and remove the retries from `destroyCoreObject`.